### PR TITLE
Users dbus api

### PR DESCRIPTION
**Pull request by jreidinger**
Ready for review. Tested using d-feet. Commit will be done as part of manager PR.

### CODE REVIEW

File: service/lib/dinstaller/dbus/users.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        @logger = logger
+        @root_password_set = false
+
+        super(PATH)
+      end
+
+      dbus_interface USERS_INTERFACE do
+        dbus_reader :root_password_set, "b"
+
+        dbus_reader :root_ssh_key, "s", dbus_name: "RootSSHKey"
```
Comment (imobachgs) [inline service/lib/dinstaller/dbus/users.rb:45]: np: What about calling it RootSSHPublicKey?

File: service/lib/dinstaller/dbus/users.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        dbus_reader :first_user, "(ssba{sv})"
+
+        dbus_method :SetRootPassword, "in Value:s, in Encrypted:b" do |value, encrypted|
+          logger.info "Setting Root Password"
+          backend.assign_root_password(value, encrypted)
+
+          self[DBus::PROPERTY_INTERFACE].PropertiesChanged(USERS_INTERFACE, {"RootPasswordSet" => !value.empty?}, [])
```
Comment (imobachgs) [inline service/lib/dinstaller/dbus/users.rb:53]: Signaling that the password is set is fine; I wonder if `SetRootPassword` should return something.

**Review by imobachgs — COMMENTED** [review]
The approach looks OK.

File: service/lib/dinstaller/dbus/users.rb [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      def root_password_set
+        backend.root_password?
+      end
+
+    private
+
+      attr_reader :logger
+
+      def backend
+        @backend = ::DInstaller::Users.instance.tap { |i| i.logger = @logger }
```
Comment (joseivanlopez) [inline service/lib/dinstaller/dbus/users.rb:91]: Why is backend assinged everytime? I would suggest to simply return the singleton instance.

File: service/lib/dinstaller/dbus/users.rb (hunk 2 above)
Comment (jreidinger) [reply service/lib/dinstaller/dbus/users.rb:53]: well, only think that comes to my mind is encrypted password, but I am not sure if we have it, as it depends on default encryption method. What else you expect from it?

File: service/lib/dinstaller/dbus/users.rb (hunk 3 above)
Comment (jreidinger) [reply service/lib/dinstaller/dbus/users.rb:91]: my fault. It should be `||=` that is also reason why I use tap to assign logger.

File: service/lib/dinstaller/dbus/users.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+        FUSER_SIG = "in FullName:s, in UserName:s, in Password:s, in AutoLogin:b, in data:a{sv}"
+        dbus_method :SetFirstUser, FUSER_SIG do |full_name, user_name, password, auto_login, data|
+          logger.info "Setting first user #{full_name}"
+          backend.assign_first_user(full_name, user_name, password, auto_login, data)
+
+          PropertiesChanged(USERS_INTERFACE, {"FirstUser" => first_user}, [])
+        end
+
+
```
Comment (ancorgs) [inline service/lib/dinstaller/dbus/users.rb:71]: At which point in time do we expect to add Rubocop? :wink:

File: service/lib/dinstaller/dbus/users.rb [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    class Users < ::DBus::Object
+      PATH = "/org/opensuse/DInstaller/Users1".freeze
+      private_constant :PATH
+
+      USERS_INTERFACE = "org.opensuse.DInstaller.Users1".freeze
+      private_constant :USERS_INTERFACE
+
+      def initialize(logger)
+        @logger = logger
+        @root_password_set = false
```
Comment (ancorgs) [inline service/lib/dinstaller/dbus/users.rb:37]: This looks like a leftover from a previous implementation. That attribute is now delegated to the backend, isn't?

File: service/lib/dinstaller/dbus/users.rb (hunk 4 above)
Comment (imobachgs) [reply service/lib/dinstaller/dbus/users.rb:71]: My main concern is not RuboCop, but RSpec ;-)

File: service/lib/dinstaller/dbus/users.rb (hunk 2 above)
Comment (imobachgs) [reply service/lib/dinstaller/dbus/users.rb]: Perhaps if it was successful or not. But I guess that it is something we should discuss for other endpoints too, so let's keep it as it is by now.

**Review by ancorgs — APPROVED** [review]
LGTM

**Pull request merged**